cmd/kubeadm/app/cmd/init.go

```diff
@@ -205,14 +204,13 @@ func NewInit(cfgPath string, cfg *kubeadmapi.MasterConfiguration, skipPreFlight
 	cfg.KubernetesVersion = ver
 	fmt.Println("Using Kubernetes version:", ver)
 
-	// TODO(phase1+) create a custom flag
+	// Warn about the limitations with the current cloudprovider solution.
 	if cfg.CloudProvider != "" {
-		if cloudprovider.IsCloudProvider(cfg.CloudProvider) {
-			fmt.Printf("cloud provider %q initialized for the control plane. Remember to set the same cloud provider flag on the kubelet.\n", cfg.CloudProvider)
-		} else {
-			return nil, fmt.Errorf("cloud provider %q is not supported, you can use any of %v, or leave it unset.\n", cfg.CloudProvider, cloudprovider.CloudProviders())
-		}
+		fmt.Println("WARNING: If you want cloudprovider integrations to work you have to set the --cloud-provider flag on ALL kubelets.")
```

Comment:
Suggest:

WARNING: For cloudprovider integrations to work --cloud-provider must be set for all kubelets in the cluster. (/etc/systemd/system/kubelet.service.d/10-kubeadm.conf can be used for this purpose)